gosamplerate.go

```diff
@@ -122,56 +137,27 @@ func (src *Src) ErrorNo() int {
 }
 
 // Simple converts a single block of samples (one or more channels) in one go. The simple API is less
-// capable than the full API (Process()). It must not be used if Audio shall be converted in junks. For full documentation see: http://www.mega-nerd.com/SRC/api_simple.html
+// capable than the full API (Process()). It must not be used if Audio shall be converted in chunks. For full documentation see: http://www.mega-nerd.com/SRC/api_simple.html
 func Simple(dataIn []float32, ratio float64, channels int, converterType int) ([]float32, error) {
-	cConverterType := C.int(converterType)
-	cChannels := C.int(channels)
-	cRatio := C.double(ratio)
-
-	dataInLength := len(dataIn)
-
-	inputBuffer := make([]C.float, dataInLength)
-	n := int(math.Ceil(ratio))
-	if n <= 0 {
-		// hack to make sure we can get the pointer to an underlying
-		// array.
-		n = 1
-	}
-	outputBuffer := make([]C.float, dataInLength*n)
-
-	// copy data into input buffer
-	for i, el := range dataIn {
-		inputBuffer[i] = C.float(el)
-	}
-
-	srcData := C.alloc_src_data()
-	defer C.free_src_data(srcData)
-
-	srcData.data_in = &inputBuffer[0]
-	srcData.data_out = &outputBuffer[0]
-	srcData.input_frames = C.long(dataInLength / channels)
-	srcData.output_frames = C.long(cap(outputBuffer) / channels)
-	srcData.src_ratio = cRatio
-
-	res := C.src_simple(srcData, cConverterType, cChannels)
-
+	if ratio <= 0 {
+		return nil, fmt.Errorf("invalid ratio %f", ratio)
```

Comment:
We should maintain the same error (libsamplerate) error message if an invalid ratio is supplied. That's also the reason the CI tests have failed. I can change this after merging your PR.